.github/workflows/main.yml

```diff
@@ -160,3 +160,10 @@ jobs:
           $env:TASK = "${{ matrix.task }}"
           conda init powershell
           & "$env:GITHUB_WORKSPACE/.ci/test_windows.ps1"
+  all-successful:
```

Comment:
@graingert FYI this check may be problematic with failures and branch protection. Here's a more comprehensive solution: https://github.com/re-actors/alls-green.

Reply:
Ah nice one!

Reply:
Could you be more specific? How might this cause problems with branch protection?

Reply:
@danepowell it's explained at https://github.com/marketplace/actions/alls-green#why but TL;DR — when some job dependencies fail, this job may end up being skipped and branch protection treats skipped jobs as success.

Reply:
I've also filed https://github.com/microsoft/LightGBM/issues/5501.